frontend/src/metabase/admin/settings/selectors.js

```diff
@@ -608,6 +595,15 @@ export const getSettings = createSelector(
     ),
 );
 
+/**
+ * @typedef {import("metabase-types/store").State} State
+ *
+ * @type {(state: State) => Record<keyof import("metabase/selectors/settings").GetSettings<State>, any>}
+ */
+export const getSettingsByKey = createSelector(getSettings, settings =>
+  _.indexBy(settings, "key"),
```

Comment:
We used `_.groupBy`, but each setting only exists once, so we can just use `_.indexBy` instead which produces an object instead of an array.